File: Laba 6/task1.cpp
#include <iostream>

int getAmountOfDigits(int number) {
    int result = 0;
    while (number != 0) {
        number /= 10;
        result++;
    }
    return result;
}

int main() {
    int number = 0, count = 0, sum = 0;
    do {
        std::cout << "Enter a number: ";
        std::cin >> number;
        count++;
        if (number >= -4 && number < 11) {
            sum += number;
        }
    } while (getAmountOfDigits(number) != 3);

    std::cout << "Amount of numbers: " << count << std::endl;
    std::cout << "Sum of numbers in [-4;11): " << sum << std::endl;

    return 0;
}

File: Laba 6/inGo/task1.go
package main

import "fmt"

func getAmountOfDigits(number int) (result int) {
	for number != 0 {
		result++
		number /= 10
	}
	return
}

func main() {
	var number, count, sum int

	for getAmountOfDigits(number) != 3 {
		fmt.Print("Enter a number: ")
		fmt.Scanln(&number)
		count++
		if number >= -4 && number < 11 {
			sum += number
		}
	}

	fmt.Printf("Amount of numbers: %v\n", count)
	fmt.Printf("Sum of numbers in [-4;11): %v\n", sum)
}
